### ISSUE TITLE
The contents of the .pyi file and the docstring do not correspond to reality.

### ISSUE THREAD

**Issue by DanisNone** [jax-ml/jax#29083]
### Description

https://github.com/jax-ml/jax/blob/50253f1acfe5434a7a50507fd940641984d9e7a7/jax/_src/numpy/ufunc_api.py#L187-L208
https://github.com/jax-ml/jax/blob/50253f1acfe5434a7a50507fd940641984d9e7a7/jax/numpy/__init__.pyi#L54-L60
https://github.com/jax-ml/jax/blob/50253f1acfe5434a7a50507fd940641984d9e7a7/jax/numpy/__init__.pyi#L83-L89

As you can see, in the first case, axis is described as a number, while in the second and third cases it is listed as an integer or None.

However, if you call, for example:

```
x = jnp.ones((1, 2, 3))
jnp.add.reduce(x, axis=(1, 2))
```
it computes correctly. Other ufuncs behave similarly, except for subtract, which raises an error.

Therefore, either:

- the docstring and .pyi file should be updated,

- or similar axis-checking logic as in subtract should be added to the other ufuncs.

### System info (python version, jaxlib version, accelerator, etc.)

jax:    0.6.1
jaxlib: 0.6.1
numpy:  2.2.5
python: 3.12.5 (tags/v3.12.5:ff3bc82, Aug  6 2024, 20:45:27) [MSC v.1940 64 bit (AMD64)]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Windows', node='Danis', release='11', version='10.0.26100', machine='AMD64')

**Comment by jakevdp** [comment]
Hi - thanks for the report! Tuple of axes works for some ufuncs but not for others: in particular, the default implementation of `reduce` raises `NotImplementedError` in the case of a tuple: https://github.com/jax-ml/jax/blob/2d4baf4772b3ad2a3832104458aa69bbf1456953/jax/_src/numpy/ufunc_api.py#L268-L270

Given that, I don't think it's appropriate to annotate `axis` as accepting `tuple`. What do you think?
